src/haz3lcore/dynamics/PatternMatch.re

```diff
@@ -266,7 +266,8 @@ and matches_cast_Sum =
   | Sequence(_, _)
   | Closure(_)
   | Cons(_)
-  | ListConcat(_) => DoesNotMatch
+  | ListConcat(_)
+  | IfThenElse(_) => DoesNotMatch
```

Comment:
elimination forms like if/then/else (and case, prj) should actually be IndetMatch. can you fix that in this PR for all of these?